### PR TITLE
do not propage config flags from group to unprovisioned device

### PR DESCRIPTION
**Pull request by AlvaroVega**
related https://github.com/telefonicaid/iotagent-node-lib/pull/1391

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,2 +1,3 @@
+- Do not propage group config (timestamp, expresisonLanguage) to autoprovision device (#1391)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]:
```suggestion
- Fix: do not propagate group config (timestamp and explicitAttrs) to autoprovisioned devices (at database level) (#1391)
```

Note that expressionLanguage was removed in a recent version, so it doens't belong to the existing API or DB model.

**Comment by fgalan** [comment]
Maybe we should tell in someplace in the documentation how this works. I mean, something like this:

> In the case of autoprovisioned devices, the associated document created in database doesn't include config group parameters (timestamp and explicitAttrs in particular). That way, these parameters can freely established modifying the device information after creation using the provisioning API. However, note that if a device (autoprovisioned or not) doesn't have these parameters defined at device level in database, the parameters are inherit from config group parameters.

Not sure where to include such fragment. Do we have some place in which autoprovisioned devices are explained? @mapedraza what do you think?

**Comment by mapedraza** [comment]
> Maybe we should tell in someplace in the documentation how this works. I mean, something like this:
> 
> > In the case of autoprovisioned devices, the associated document created in database doesn't include config group parameters (timestamp and explicitAttrs in particular). That way, these parameters can freely established modifying the device information after creation using the provisioning API. However, note that if a device (autoprovisioned or not) doesn't have these parameters defined at device level in database, the parameters are inherit from config group parameters.
> 
> Not sure where to include such fragment. Do we have some place in which autoprovisioned devices are explained? @mapedraza what do you think?

I agree, it seems reasonable. Maybe a new chapter explaining autoprovision under https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#iot-agent-information-model

**Comment by mapedraza** [comment]
> Could you add such documentation to PR [telefonicaid/iotagent-node-lib#1391](https://github.com/telefonicaid/iotagent-node-lib/pull/1391), pls?

Yes, I opened an issue but I will add to that PR

**Comment by fgalan** [comment]
> > Could you add such documentation to PR [telefonicaid/iotagent-node-lib#1391](https://github.com/telefonicaid/iotagent-node-lib/pull/1391), pls?
> 
> Yes, I opened an issue but I will add to that PR

Taking into account the modification to the documentation is mall, I'd suggest in the PR we have now on the table (ideally, we should should include code + test + doc in the same PR, only when this cannot be done we use "debt" issues).

I have found another section that can be relevant: https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#autoprovision-configuration-autoprovision

**Comment by mapedraza** [comment]
> > > Could you add such documentation to PR [telefonicaid/iotagent-node-lib#1391](https://github.com/telefonicaid/iotagent-node-lib/pull/1391), pls?
> > 
> > 
> > Yes, I opened an issue but I will add to that PR
> 
> Taking into account the modification to the documentation is mall, I'd suggest in the PR we have now on the table (ideally, we should should include code + test + doc in the same PR, only when this cannot be done we use "debt" issues).
> 
> I have found another section that can be relevant: https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#autoprovision-configuration-autoprovision

Done here: 
https://github.com/telefonicaid/iotagent-node-lib/pull/1391/commits/64f4ba31544fc48de7ad11d7eb373ec8e9f81735

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**